### ISSUE TITLE
[BUG] Save as dialogs created by Qt starts search when trying to enter filename

### ISSUE THREAD

**Issue by gregersn** [linuxmint/cinnamon#11078]
```
 * Cinnamon version 5.2.7
 * Distribution - Fedora 36
 * Graphics hardware: Geforce 2060, Nvidia proprietary driver, 515.65.01
 * 64 bit
 ```

### Issue

When a "Save As" file dialog opens, and I start entering a filename, functions trigger, and makes it impossible.

### Steps to reproduce

- Start program that uses Qt, VLC used for the rest of the example here
- Click Media -> Save Playlist to File
- Type filename in "name"
- "Search" field pops up and catches all input.

### Expected behaviour

Be able to enter a filename so I can save files.

### Other information

Have tested in Gnome on the same machine, where issue does not occur.

**Comment by Fool4UAnyway** [comment]
I noticed the same behavior in another Save (As) dialog, in the Brave web browser in this case.

I do understand the behavior, but it contains another bug.

When entering another character, the search results may be restricted (less matches).

When removing a character, the search may show more results (again).

This is true, except for the last character. If the search box is empty, no more files are shown than with the last character that was removed. This is a bug in my view. I will create an issue for this, at it is also the case in File Open dialogs.

Not all Save As dialogs show this behavior. Others respond as OP expects.
